python/cudf/cudf/core/dtypes.py

```diff
@@ -221,6 +221,7 @@ class DecimalDtype(ExtensionDtype):
 
     name = "decimal"
     _metadata = ("precision", "scale")
+    _MAX_PRECISION = len(str(np.iinfo("int64").max)) - 1
```

Comment:
Is there a reason you used this approach instead of the `log` approach?

Reply:
Not particularly -- I don't have a preference. Can use the `log` based approach if that seems clearer

Reply:
Fixed to use log-based approach